framework/tlvf/tlvf.py

```diff
@@ -874,6 +881,12 @@ def addClassParam(self, obj_meta, param_name, param_type, param_type_info, param
                 if TypeInfo(param_type).type == TypeInfo.CLASS: #TODO:only if it's the last member of the class
                     lines_h.append( "std::shared_ptr<%s> create_%s();" % (param_type, param_name) ) #TODO: maybe change to 'create' and add '_entry' postfix
                     lines_cpp.append( "std::shared_ptr<%s> %s::create_%s() {" % (param_type, obj_meta.name, param_name) )
+                    lines_cpp.append( "%sif (m_%s__ > %s) {" %(self.getIndentation(1), self.MEMBER_LOCK_ORDER_COUNTER, param_meta.list_index) )
+                    lines_cpp.append( "%sTLVF_LOG(ERROR) << \"Out of order allocation for variable length list %s, abort!\";" %(self.getIndentation(2), param_name) )
+                    lines_cpp.append( "%sreturn nullptr;" %self.getIndentation(2))
+                    lines_cpp.append( "%s} else {" %self.getIndentation(1))
+                    lines_cpp.append( "%sm_%s__ = %s;" %(self.getIndentation(2), self.MEMBER_LOCK_ORDER_COUNTER, param_meta.list_index) )
```

Comment:
Might be better to first make all sanity tests (count == 0 and available space on buffer) and only then update the counter.